examples/ACE/utils/NNIAP.jl

```diff
@@ -42,22 +62,60 @@ end
 
 
 # Loss function ################################################################
-function loss(nn, iap, ds, w_e = 1, w_f = 1)
+function loss(nn, iap, ds, w_e::Real=1, w_f::Real=1)
     nniap = NNIAP(nn, iap)
     es, es_pred = get_all_energies(ds), get_all_energies(ds, nniap)
     fs, fs_pred = get_all_forces(ds), get_all_forces(ds, nniap)
     return w_e * Flux.mse(es_pred, es) + w_f * Flux.mse(fs_pred, fs)
 end
 
 
+function loss(nn, iap,  batch, true_energy, local_descriptors, w_e=1, w_f=1)
+    # nniap = NNIAP(nn, iap)
+    es_pred = sum(sum(nn.(local_descriptors)))
+    # fs = get_all_forces(batch)
+    # fs_pred = get_all_forces(batch, nn , local_descriptors, _device)
+    return w_e * Flux.mse(es_pred, true_energy)  # +   w_f * Flux.mse(fs_pred, fs)
+end
+
```

Comment:
`e_pred` calculation needs to be fixed:
 
`local_descriptors` is a column-major ordered matrix containing the local descriptors of a set of configurations. E.g., suppose the first configuration c1 has N atoms, and the second configuration c2 has M atoms, then:
`local_descriptors = { ld_atom1_c1, ld_atom2_c1, ..., ld_atomN_c1, ld_atom1_c2, ... ld_atomM_c2, ... }`

where `ld_atomi_cj `is a column vector. In the function `learn!` these vectors where stacked into a matrix and transfered to the GPU.

`e_pred` is a vector of predicted energies associated to a particular batch of configurations (the batch was defined in the function `learn!`). 

The predicted energy of a configuration c (that is, one of the elements of `e_pred`) is defined as:
`e_pred_c = sum_i nn(local_descriptors_i)`
where i represents the atoms of the configuration c

The code `sum(sum(nn.(local_descriptors)))` does not implement the definition above. 

An option is to first apply the nn model to the matrix of local descriptors (`nn.(local_descriptors)`). Them, split the result (that is a CuArray) with respect to the number of atoms of each configuration in the local descriptor matrix (probably you need a new vector with this information). Finally, sum each part to obtain each predicted energy.